include/swift/SIL/SILNode.h

```diff
@@ -98,6 +98,11 @@ class alignas(8) SILNode {
     Instruction
   };
 
+  enum SILNodeIsCanonical {
+    No = 0,
+    Yes = 1,
+  };
```

Comment:
I just realized this PR is causing the "canonical SILNode" misnomer to rapidly metastasize.

The name is conceptually confusing. What the API calls a canonical SILNode is used to identify the representative instruction. The representative SILNode is *not* the canonical form of a non-representative SILNode. They are simply different entities! So just call it the representative SILNode.

Reply:
I wasn't sure whether to use the _t prefix like we do in other places. The big thing I wanted to avoid here is just using a boolean. This is a really important value in our model and making it obvious where we are using it is important.

Reply:
I agree with Andy that "representative" is a better term here, since there's a semantic difference.

I think the Yes/No thing is fine.  This should be an ``enum class``, though.  The existing pattern would be something like ``enum IsRepresentative_t : bool { IsNotRepresentative, IsRepresentative };"``, but I'm not sure that's actually a *good* pattern.